File: Android/melody-runtime/src/main/java/com/melody/runtime/components/MelodyDisclosure.kt
package com.melody.runtime.components

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ExpandLess
import androidx.compose.material.icons.filled.ExpandMore
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.melody.core.schema.ComponentDefinition

@Composable
fun MelodyDisclosure(
    definition: ComponentDefinition,
    resolvedLabel: String,
    content: @Composable () -> Unit
) {
    var expanded by remember { mutableStateOf(false) }

    Column(modifier = Modifier.melodyStyle(definition.style)) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            modifier = Modifier
                .fillMaxWidth()
                .clickable { expanded = !expanded }
                .padding(vertical = 8.dp)
        ) {
            Text(resolvedLabel, modifier = Modifier.weight(1f))
            Icon(
                imageVector = if (expanded) Icons.Default.ExpandLess else Icons.Default.ExpandMore,
                contentDescription = if (expanded) "Collapse" else "Expand"
            )
        }

        AnimatedVisibility(visible = expanded) {
            Column {
                content()
            }
        }
    }
}

File: Sources/Runtime/Components/MelodyDisclosure.swift
import SwiftUI
import Core

struct MelodyDisclosure<Content: View>: View {
    let definition: ComponentDefinition
    let resolvedLabel: String
    @ViewBuilder let content: () -> Content

    var body: some View {
        #if os(tvOS)
        VStack {
            content()
        }
        .melodyStyle(definition.style)
        #else
        DisclosureGroup(resolvedLabel) {
            content()
        }
        .melodyStyle(definition.style)
        #endif
    }
}
